aten/src/ATen/native/mps/operations/UnaryKernel.mm

```diff
@@ -0,0 +1,133 @@
+  auto rc =
+      [device newLibraryWithSource:[NSString stringWithUTF8String:fmt::format(UNARY_KERNEL_TEMPLATE, t1, t2).c_str()]
+                           options:options
+                             error:&error];
+  TORCH_CHECK(rc != nil && error == nil, "Failed to compile library: ", [[error localizedDescription] UTF8String]);
+  libMap[key] = rc;
+  return rc;
+}
+
+static id<MTLComputePipelineState> getCPLState(id<MTLDevice> device,
+                                               const std::string& t1,
+                                               const std::string& t2,
+                                               const std::string& fname) {
+  auto key = t1 + t2 + fname;
+  static std::unordered_map<std::string, id<MTLComputePipelineState>> cplMap;
+  auto it = cplMap.find(key);
+  if (it != cplMap.end()) {
+    return it->second;
+  }
+  NSError* error = nil;
+  auto library = compileUnaryOpsLibrary(device, t1, t2);
+  id<MTLFunction> func = [library newFunctionWithName:[NSString stringWithUTF8String:fname.c_str()]];
+  TORCH_CHECK(func != nil, "Can't get function ", fname);
+  auto rc = [device newComputePipelineStateWithFunction:func error:&error];
+  TORCH_CHECK(
+      rc != nil && error == nil, "Failed to construct pipeline state: ", [[error localizedDescription] UTF8String]);
+  cplMap[key] = rc;
+  return rc;
+}
+
+TORCH_IMPL_FUNC(erfinv_out_mps)(const Tensor& self, const Tensor& output_) {
+  // handle erfinv ops using metal kernel
+  // erfinv algorithm ported from aten/src/ATen/native/Math.h
+  // https://github.com/pytorch/pytorch/blob/4154c8ea159fdaecc71ee9af820ac956193c875b/aten/src/ATen/native/Math.h#L152
+
+  TORCH_CHECK(self.scalar_type() != ScalarType::Double, "MPS does not support erfinv op with scalar type: Double");
+
+  Tensor outputTensor = output_;
+  bool needs_output_copy = false;
+  uint32_t length = output_.numel();
+  if (length == 0) {
+    return;
+  }
+  using namespace mps;
+  @autoreleasepool {
+    Tensor inputTensor = self;
+    id<MTLDevice> device = MPSDevice::getInstance()->device();
+    id<MTLComputePipelineState> cplState =
+        getCPLState(device, getMetalType(outputTensor), getMetalType(self), "erfinv_mps_kernel");
+
+    if (!self.is_contiguous()) {
+      inputTensor = inputTensor.contiguous();
+      outputTensor = outputTensor.contiguous();
+      needs_output_copy = true;
+    }
+
+    MPSStream* mpsStream = getCurrentMPSStream();
+    dispatch_sync(mpsStream->queue(), ^() {
+      id<MTLComputeCommandEncoder> computeEncoder = mpsStream->commandEncoder();
+      id<MTLBuffer> outBuf = getMTLBufferStorage(outputTensor);
+      id<MTLBuffer> inputBuf = getMTLBufferStorage(inputTensor);
+
+      getMPSProfiler().beginProfileKernel(cplState, "erf_inv", {self});
+
+      [computeEncoder setComputePipelineState:cplState];
+      [computeEncoder setBuffer:outBuf offset:0 atIndex:0];
+      [computeEncoder setBuffer:inputBuf offset:0 atIndex:1];
```

Comment:
@TaiPhamD Sorry, I just noticed that the offset here should be the tensor's storage offset in bytes, i.e. `tensor.storage_offset() * tensor.element_size()`, instead of zero. Can you please submit a follow-up PR for it.

In PyTorch we use storage offset to achieve efficient tensor slicing, and hard-cording zero here would lead to incorrect results when the input or output tensor is sliced, i.e. having storage offset.

You could use the following snippet to verify, `y` and `y2` should be the same:

```python
def test_erfinv():
    import torch
    for device in ['cpu', 'mps']:
        x = torch.tensor([0.1, 0.2, 0.3, 0.4, 0.5], device=device)
        y = x[2:].erfinv()

        x2 = torch.tensor([0.3, 0.4, 0.5], device=device)
        y2 = x2.erfinv()

        print(y)
        print(y2)

        torch.testing.assert_close(y, y2)
        print(f"{device} passes.")
```

Reply:
Good catch @qqaatw , do you want to follow up on this PR?

Reply:
@kulinseth, @TaiPhamD has submitted a follow-up PR here: https://github.com/pytorch/pytorch/pull/105801